subsys/mpsl/cx/bluetooth/mpsl_cx_1w_bluetooth.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ *   This file implements a generic bluetooth external radio 1w coexistence
+ *   interface.
+ */
+#include <stddef.h>
+#include <stdint.h>
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <devicetree.h>
+#include <nrfx.h>
+#include <nrfx_ppi.h>
+#include <nrfx_gpiote.h>
+#include <mpsl_coex.h>
+
+#define COEX_TIMER NRF_TIMER1
+
+#if DT_HAS_COMPAT_STATUS_OKAY(sdc_radio_coex_one_wire)
+#define COEX_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(sdc_radio_coex_one_wire)
+#else
+#define COEX_NODE DT_INVALID_NODE
+#error No "sdc-radio-coex-one-wire"-compatible (and enabled) nodes found
+#endif
+
+#if !IS_ENABLED(CONFIG_SOC_SERIES_NRF52X)
+#error Bluetooth coex is only supported on the nRF52 series
+#endif
+
+#define MPSL_COEX_BT_GPIO_POLARITY_GET(dt_property)                                                \
```

Comment:
we could, but I would prefer to have it in this file, then it is clearer to look at the 3W part, in my opinion. I think you would then have all the configuration in one place instead of having half of it in the header an the other half in the c file.

Reply:
Then we would need to define in the C file the COEX_NODE first before including the header. I think that's less clean.

Reply:
alright, no strong opinion :)